packages/devtools/lib/src/utils.dart

```diff
@@ -117,6 +117,28 @@ void executeWithDelay(Duration delay, void callback(),
   }
 }
 
+String longestFittingSubstring(
+  String originalText,
+  num maxWidth,
+  List<num> asciiMeasurements,
+) {
+  if (originalText.isEmpty) return originalText;
+
+  num currentWidth = 0;
+  int longestIndex = 0;
+  final charCodes = ascii.encode(originalText);
+  for (int i = 0; i < charCodes.length; i++) {
```

Comment:
simpler: initialize i outside the loop. then you can eliminate the longestIndex field as it is just i
and also eliminate the +1 in the substring method call.

Reply:
done.